sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -231,6 +234,15 @@ case class Alias(child: Expression, name: String)(
     copy(child = newChild)(exprId, qualifier, explicitMetadata, nonInheritableMetadataKeys)
 }
 
+// Singleton tree pattern BitSet for all AttributeReference instances.
+object AttributeReferenceTreeBits {
+  val bits: BitSet = {
+    val bits: BitSet = new BitSet(TreePattern.maxId)
+    bits.set(ATTRIBUTE_REFERENCE.id)
+    bits
```

Comment:
Do we consider having a new class for immutable BitSet to avoid the singleton bits being modified?

Reply:
+1

Reply:
Done.